pkg/kubectl/cmd/macro/macro.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package macro
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+// commandParamVisitorFunc has an opportunity to visit each parameter in specified command
+type commandParamVisitorFunc func(cmd *cobra.Command, params map[string]interface{})
+
+// commandParamVisitorFunc modifies the parameter value for each enumerated
+// parameter to be a string slice rather than a single string.  useful for
+// command parameters that may be specified multiple times.
+func commandParamStringSliceVistor(stringSliceParams []string) commandParamVisitorFunc {
```

Comment:
@pmorie - typos on func names are never nits!  they are the worst thing ever.  thanks for catching.